### PR TITLE
WASM - New wabt version for spec test

### PR DESCRIPTION
**Pull request by Cellule**
Update test runner to work with the new output of wabt.
This includes support for a lot of invalid cases.
There are some errors regaring kinking error that will be fixed in a later PR to add the new WebAssembly.LinkError type

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/chakracore/2208)

### CODE REVIEW

**Comment by Cellule** [comment]
@MikeHolman Can you review this please ?
I highly recommend to review only the first 2 commits as they're the ones that actually contains changes.
The third commit is the regeneration of the tests

**Comment by Cellule** [comment]
ping

**Comment by Cellule** [comment]
ping

File: test/WasmSpec/spec.js [hunk 1]
```diff
@@ -2,186 +2,361 @@
 // Copyright (C) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
 //-------------------------------------------------------------------------------------------------------
-// This file has been modified by Microsoft on [07/2016].
-
-/*
- * Copyright 2016 WebAssembly Community Group participants
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (MikeHolman) [inline test/WasmSpec/spec.js]: I think it's safest to keep the copyright

**Pull request merged**